packages/angular/cli/src/command-builder/command-runner.ts

```diff
@@ -54,18 +55,26 @@ const COMMANDS = [
   UpdateCommandModule,
   RunCommandModule,
   CacheCommandModule,
+  CompletionCommandModule,
 ].sort(); // Will be sorted by class name.
 
 const yargsParser = Parser as unknown as typeof Parser.default;
 
 export async function runCommand(args: string[], logger: logging.Logger): Promise<number> {
   const {
     $0,
-    _: positional,
+    _,
     help = false,
     jsonHelp = false,
+    getYargsCompletions = false,
     ...rest
-  } = yargsParser(args, { boolean: ['help', 'json-help'], alias: { 'collection': 'c' } });
+  } = yargsParser(args, {
+    boolean: ['help', 'json-help', 'get-yargs-completions'],
+    alias: { 'collection': 'c' },
+  });
+
+  // When `getYargsCompletions` is true the scriptName 'ng' at index 0 is not removed.
+  const positional = getYargsCompletions ? _.slice(1) : _;
```

Comment:
Is this an intentional `yargs` behavior or something that might be _fixed_ in the future?

Reply:
I think this is somewhat expected because of the order of args.


```
[
    '...nvm/versions/node/v14.18.1/bin/node',
    '....nvm/versions/node/v14.18.1/bin/ng',
    '--get-yargs-completions',
    'ng',
    'b',
    '-'
  ],
```